packages/util/src/requests.ts

```diff
@@ -0,0 +1,28 @@
+import { concatBytes } from 'ethereum-cryptography/utils'
+
+export type RequestBytes = Uint8Array
+
+export interface RequestData {
+  type: number
+  data: Uint8Array
+}
+
+export interface CLRequestType {
+  readonly type: number
+  readonly bytes: Uint8Array
+  serialize(): Uint8Array
+}
+
+export class CLRequest implements CLRequestType {
+  type: number
+  bytes: Uint8Array = new Uint8Array()
+  constructor(type: number, bytes: Uint8Array) {
+    if (type === undefined) throw new Error('request type is required')
+    this.type = type
+    this.bytes = bytes
+  }
+
+  serialize() {
+    return concatBytes(Uint8Array.from([this.type]), this.bytes)
```

Comment:
Nit: this does not work for types > 255. (But since we are super far from having to support this it is fine)

Reply:
Not disagreeing with you but the EIP specified the first byte as the type byte, so this is according to spec.

Reply:
Ah good point, the EIP itself does not support types > 255. Will resolve.